hadoop-hdds/framework/src/test/java/org/apache/hadoop/hdds/security/x509/certificate/authority/TestDefaultCAServer.java

```diff
@@ -361,12 +357,11 @@ public void testInitWithCertChain(@TempDir Path tempDir) throws Exception {
           java.sql.Date.valueOf(beginDate), java.sql.Date.valueOf(endDate), csr,
           scmId, clusterId, String.valueOf(System.nanoTime()));
       CertificateFactory certFactory = CertificateFactory.getInstance("X.509");
-      CertificateCodec certificateCodec = new CertificateCodec(securityConfig,
-          scmCertificateClient.getComponentName());
 
       CertPath certPath = certFactory.generateCertPath(ImmutableList.of(signedCert, externalCert));
-      certificateCodec.writeCertificate(tempDir, externalCaCertFileName,
-          CertificateCodec.getPEMEncodedString(certPath));
+      CertificateCodec certificateCodec = securityConfig.getCertificateCodec();
+      certificateStorage.writeCertificate(Paths.get(tempDir.toString(), externalCaCertFileName),
+          certificateCodec.getPEMEncodedString(certPath));
```

Comment:
Wouldn't it be better to do the conversion within the storage? In that case we don't need to query the codec here either. I have seen a couple of other instances where the codec is used before calling the storage, and I have the feeling that the storage should accept the CertPath, and use the codec internally to create the PEM format. That way we encapsulate the decision about the format also.

This change would change the notion of a few other tests and probably somewhat the production code also, so please look at similar usages also.